File: Cosmos-Android/app/src/main/java/wannabit/io/cosmostaion/model/kava/HardMyBorrow.java
package wannabit.io.cosmostaion.model.kava;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

import wannabit.io.cosmostaion.model.type.Coin;

public class HardMyBorrow {
    @SerializedName("borrower")
    public String borrower;

    @SerializedName("amount")
    public ArrayList<Coin> amount;

    @SerializedName("index")
    public ArrayList<HardIndex> index;
}

File: Cosmos-IOS/Cosmostation/Cosmostation/Model/KavaModel/HardMyBorrow.swift
//
//  HardMyBorrow.swift
//  Cosmostation
//
//  Created by 정용주 on 2021/03/04.
//  Copyright © 2021 wannabit. All rights reserved.
//

import Foundation

public struct HardMyBorrow {
    var borrower: String?
    var amount: Array<Coin>?
    var index: Array<HardIndex>?
    
    init(_ dictionary: NSDictionary?) {
        self.borrower = dictionary?["borrower"] as? String
        if let rawAmounts = dictionary?["amount"] as? Array<NSDictionary> {
            self.amount = Array<Coin>()
            for rawAmount in rawAmounts {
                self.amount?.append(Coin.init(rawAmount))
            }
        }
        if let rawHardIndexs = dictionary?["index"] as? Array<NSDictionary> {
            self.index = Array<HardIndex>()
            for rawHardIndex in rawHardIndexs {
                self.index?.append(HardIndex.init(rawHardIndex))
            }
        }
    }
}
